rpc/src/v1/impls/eth_signing.rs

```diff
@@ -24,25 +24,40 @@ use util::numbers::*;
 use util::keys::store::AccountProvider;
 use v1::helpers::{SigningQueue, ConfirmationsQueue};
 use v1::traits::EthSigning;
-use v1::types::TransactionRequest;
+use v1::types::{TransactionRequest, Bytes};
 use v1::impls::{sign_and_dispatch, error_codes};
 
 
 /// Implementation of functions that require signing when no trusted signer is used.
-pub struct EthSigningQueueClient {
+pub struct EthSigningQueueClient<M: MinerService> {
 	queue: Weak<ConfirmationsQueue>,
+	miner: Weak<M>,
 }
 
-impl EthSigningQueueClient {
+impl<M: MinerService> EthSigningQueueClient<M> {
 	/// Creates a new signing queue client given shared signing queue.
-	pub fn new(queue: &Arc<ConfirmationsQueue>) -> Self {
+	pub fn new(queue: &Arc<ConfirmationsQueue>, miner: &Arc<M>) -> Self {
 		EthSigningQueueClient {
 			queue: Arc::downgrade(queue),
+			miner: Arc::downgrade(miner),
+		}
+	}
+
+	fn fill_optional_fields(&self, miner: Arc<M>, mut request: TransactionRequest) -> TransactionRequest {
+		if let None = request.gas {
+			request.gas = Some(miner.sensible_gas_limit());
```

Comment:
The logic here is the same as in `impls/mod.rs#sign_and_dispatch`, using signer does not change this behaviour.
Replacing this can go in a separate PR.